### ISSUE TITLE
[quotes] False positive requiring backticks in types

### ISSUE THREAD

**Issue by BBosman** [typescript-eslint/typescript-eslint#756]
**Repro**

```JSON
{
  "rules": {
    "quotes": ["error", "backtick", { "avoidEscape": true }]
  }
}
```

```TS
class MockObserver {
  public hasSubscribers: IBindingTargetObserver['hasSubscribers'];
}
```

**Expected Result**
No rule violations as those can't be backticks.

**Actual Result**
Violation of `quotes`.

**Additional Info**

Actually changing them to backticks gives a compiler error.
```
public hasSubscribers: IBindingTargetObserver[`hasSubscribers`];
```
Error: `']' expected`

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `1.12.0` |
| `@typescript-eslint/parser`        | `1.12.0` |
| `TypeScript`                       | `3.5.3` |
| `ESLint`                           | `5.16.0` |
| `node`                             | `10.15.0` |
| `npm`                              | `6.10.2` |

**Comment by bradzacher** [comment]
See additional request for quotes rule in: #757
